packages/node/src/request-handler.ts

```diff
@@ -127,8 +127,20 @@ export class RequestHandler {
   }
 
   public async getSigner(): Promise<Signer> {
-    return this.provider instanceof JsonRpcProvider
-      ? await this.provider.getSigner()
-      : this.wallet;
+    try {
+      const signer = await (this.provider as JsonRpcProvider).getSigner();
+      await signer.getAddress();
```

Comment:
Why are we awaiting for this here?

Reply:
Using a `JsonRpcProvider` instead of an `InfuraProvider` or a `BaseProvider` means that its hard to differentiate between `localhost:8545` and `ropsten.infura.io/metamask` so calling `getAddress` throws an error (`UNSUPPORTED_OPERATION`)